vaadin-combo-box-flow-demo/src/main/java/com/vaadin/flow/component/combobox/demo/ComboBoxView.java

```diff
@@ -254,29 +260,107 @@ private void storingCustomValues() {
+        //@formatter:off
+        // begin-source-example
+        // source-example-heading: Lazy Loading with Exact Items Count
         // PersonService can be found:
         // https://github.com/vaadin/vaadin-combo-box-flow/tree/master/vaadin-combo-box-flow-demo/src/main/java/com/vaadin/flow/component/combobox/demo/service/PersonService.java
 
         ComboBox<Person> comboBox = new ComboBox<>();
         PersonService service = new PersonService();
         /*
-         * This data provider doesn't load all the items to the server memory
-         * right away. The component calls the first provided callback to fetch
-         * items from the given range with the given filter. The second callback
-         * should provide the number of items that match the query.
+         * By using these callbacks the ComboBox doesn't load all the items to
+         * the server memory right away. The ComboBox calls the first provided
+         * callback to fetch items from the given range with the given filter.
+         * The second callback should provide the number of items that match
+         * the query.
+         */
+        comboBox.setItems(
+                query -> service.fetch(query.getFilter().orElse(null),
+                        query.getOffset(), query.getLimit()),
+                query -> service.count(query.getFilter().orElse(null)));
+        // end-source-example
+        //@formatter:on
+        comboBox.setId("with-exact-items-count");
+        addCard("Lazy Loading", "Lazy Loading with Exact Items Count",
+                comboBox);
+    }
+
+    private void lazyLoadingWithCustomItemCountEstimate() {
+        // @formatter:off
+        // begin-source-example
+        // source-example-heading: Fast Scroll with Custom Item Count Estimate
+        // PersonService can be found:
+        // https://github.com/vaadin/vaadin-combo-box-flow/tree/master/vaadin-combo-box-flow-demo/src/main/java/com/vaadin/flow/component/combobox/demo/service/PersonService.java
+        // The backend will have 12345 items
+        PersonService service = new PersonService(12345);
+        ComboBox<Person> comboBox = new ComboBox<>();
+
+        ComboBoxLazyDataView<Person> lazyDataView = comboBox.setItems(
+                query -> service.fetch(query.getFilter().orElse(null),
+                        query.getOffset(), query.getLimit()));
+
+        /*
+         * By default, the combo box will initially adjust the scrollbar to 200
+         * items and as the user scrolls down it automatically increases the
+         * size by 200 until the backend runs out of items.
+         *
+         * Both the estimated item count and its increase can be customized to
+         * allow the user to scroll down faster when the backend will have a lot
+         * of items.
          */
-        comboBox.setDataProvider(service::fetch, service::count);
+        lazyDataView.setItemCountEstimate(1000);
+        lazyDataView.setItemCountEstimateIncrease(1000);
```

Comment:
I would keep this paragraph "By default, the combo box will initially adjust the scrollbar to 200...".

Increase of 100 is less than the default one of 200, and this may confuse the users, because the showcase of this demo is a fast scrolling and the question then would be "How 100 can be faster than 200", even the page size is decreased comparing to default one.

Reply:
You missed the point - this is not grid and since it has filtering that the user should use the expectancy is that there will be fewer items and less scrolling so the demo would be instead to showcase the customizability in a use case, that I think, is more natural for CB.

Reply:
Got it, thanks! So I modified it as follows:
```
        /*
         * By default, the combo box will initially adjust the scrollbar to 200
         * items and as the user scrolls down it automatically increases the
         * size by 200 until the backend runs out of items.
         *
         * In case of huge number of items in the backend, combo box's text
         * filter can be used to decrease the number of shown items and
         * you can customize the estimated item count and its increase to allow
         * the user scroll through the filtered items set.
         */
        comboBox.setPageSize(20);
        lazyDataView.setItemCountEstimate(100);
        lazyDataView.setItemCountEstimateIncrease(100);
```